roomserver/internal/perform/perform_create_room.go

```diff
@@ -114,23 +114,34 @@ func (c *Creator) PerformCreateRoom(ctx context.Context, userID spec.UserID, roo
 	}
 
 	var guestsCanJoin bool
-	switch createRequest.StatePreset {
-	case spec.PresetPrivateChat:
-		joinRuleContent.JoinRule = spec.Invite
-		historyVisibilityContent.HistoryVisibility = historyVisibilityShared
-		guestsCanJoin = true
-	case spec.PresetTrustedPrivateChat:
-		joinRuleContent.JoinRule = spec.Invite
-		historyVisibilityContent.HistoryVisibility = historyVisibilityShared
-		for _, invitee := range createRequest.InvitedUsers {
-			powerLevelContent.Users[invitee] = 100
+	if createRequest.StatePreset == "" {
+		switch createRequest.Visibility {
+		case "private", "":
+			joinRuleContent.JoinRule = spec.Invite
+			historyVisibilityContent.HistoryVisibility = historyVisibilityShared
+			guestsCanJoin = true
+		case "public":
+			joinRuleContent.JoinRule = spec.Public
+			historyVisibilityContent.HistoryVisibility = historyVisibilityShared
```

Comment:
I would prefer it if you just set the `createRequest.StatePreset` rather than re-implement the state preset code here.

i.e:
```go
// If unspecified, the server should use the visibility
// to determine which preset to use.
// A visibility of public equates to a preset of public_chat
// and private visibility equates to a preset of private_chat.
if createRequest.StatePreset == "" {
    switch createRequest.Visibility {
        // Rooms default to private visibility if this key is not included.
        case "private", "":
            createRequest.StatePreset = "private_chat"
        case "public":
            createRequest.StatePreset = "public_chat"
    }
}

switch createRequest.StatePreset {
// ... rest of code

```